hack/bootstrap-kubestellar.sh

```diff
@@ -0,0 +1,263 @@
+      aarch64*) echo "arm64" ;;
+      *)        echo "Unsupported architecture type: $HOSTTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_full_path() {
+    echo "$(cd "$1"; pwd)"
+}
+
+kcp_version=""
+kubestellar_version=""
+os_type=""
+arch_type=""
+folder=""
+strip_bin="false"
+verbose=""
+kcp_address=""
+kubestellar_imw=""
+kubestellar_wmw=""
+
+while (( $# > 0 )); do
+    case "$1" in
+    (--kcp-version)
+        if (( $# > 1 ));
+        then { kcp_version="$2"; shift; }
+        else { echo "$0: missing release version" >&2; exit 1; }
+        fi;;
+    (--kubestellar-version)
+        if (( $# > 1 ));
+        then { kubestellar_version="$2"; shift; }
+        else { echo "$0: missing release version" >&2; exit 1; }
+        fi;;
+    (--os)
+        if (( $# > 1 ));
+        then { os_type="$2"; shift; }
+        else { echo "$0: missing OS type" >&2; exit 1; }
+        fi;;
+    (--arch)
+        if (( $# > 1 ));
+        then { arch_type="$2"; shift; }
+        else { echo "$0: missing architecture type" >&2; exit 1; }
+        fi;;
+    (--bind)
+        if (( $# > 1 ));
+        then { kcp_address="$2"; shift; }
+        else { echo "$0: missing ip address" >&2; exit 1; }
+        fi;;
+    (--imw)
+        if (( $# > 1 ));
+        then { kubestellar_imw="$2"; shift; }
+        else { echo "$0: missing workspace name" >&2; exit 1; }
+        fi;;
+    (--wmw)
+        if (( $# > 1 ));
+        then { kubestellar_wmw="$2"; shift; }
+        else { echo "$0: missing workspace name" >&2; exit 1; }
+        fi;;
+    (--folder)
+        if (( $# > 1 ));
+        then { folder="$2"; shift; }
+        else { echo "$0: missing installation folder" >&2; exit 1; }
+        fi;;
+    (--verbose|-V)
+        verbose="-V";;
+    (-h|--help)
+        echo "Usage: $0 [--kcp-version release_version] [--kubestellar-version release_version] [--os linux|darwin] [--arch amd64|arm64] [--folder installation_folder] [--strip-bin] [-V|--verbose]"
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2 ; exit 1;
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2; exit 1;
+        exit 1;;
+    esac
+    shift
+done
+
+if [ "$kcp_version" == "" ]; then
+    kcp_version=$(kcp_get_latest_version)
+fi
+
+if [ "$kubestellar_version" == "" ]; then
+    kubestellar_version=$(kubestellar_get_latest_version)
+fi
+
+if [ "$os_type" == "" ]; then
+    os_type=$(get_os_type)
+fi
+
+if [ "$arch_type" == "" ]; then
+    arch_type=$(get_arch_type)
+fi
+
+if [ "$folder" == "" ]; then
+    folder="$PWD"
+fi
+
+# Ensure kcp is installed
+if [ "$(kcp_installed)" == "false" ]; then
+    if [ "$verbose" != "" ]; then
+        echo "Installing kcp..."
+    fi
+    bash <(curl -s https://raw.githubusercontent.com/kcp-dev/edge-mc/main/hack/install-kcp-with-plugins.sh) --version $kcp_version --os $os_type --arch $arch_type --folder  $folder/kcp --create-folder $verbose
+    if [[ ! ":$PATH:" == *":$(get_full_path $folder/kcp/bin):"* ]]; then
```

Comment:
This does not consider the possibility of strip-bin.  I would be happy if that option were gone.

Reply:
yes, currently it's not used